viewing_party/party.py

```diff
@@ -1,23 +1,148 @@
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+ 
+def get_watched_avg_rating(user_data): 
+    sum = 0
+    average = 0 
+    try:
+        for movie in user_data['watched']: 
+            sum += movie['rating']
+        average = sum / len(user_data['watched'])
+        return average
+    except: 
+        return 0.0
+            
+    
+def get_most_watched_genre(user_data):
+    genre_list = []
+    try: 
+        for movie in user_data['watched']:
+            genre_list.append(movie['genre'])
+
+        genre_count = Counter(genre_list)
+        fave_genre = ((genre_count.most_common(1))[0][0])
+        return fave_genre   
+    except: 
+        return None 
+    
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def friend_watched_list(user_data):
+    friend_watched = []     
+    for friend in user_data['friends']: 
+        for a_movie in friend['watched']:
+            if a_movie in friend_watched: 
+                continue 
+            else: 
+                friend_watched.append(a_movie)
+    return friend_watched 
+
+
+def get_unique_watched(user_data):
+    friend_watched = friend_watched_list(user_data)
+    user_unique = []
+   
+    for movie in user_data['watched']: 
+        if movie not in friend_watched: 
+            user_unique.append(movie)
+
+    return(user_unique)
+
+
+def get_friends_unique_watched(user_data): 
+    friend_watched = friend_watched_list(user_data)
+    friend_unique = []  
+    for movie in friend_watched: 
+        if movie not in user_data['watched']: 
+            friend_unique.append(movie)
+
+    return(friend_unique)
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    user_subscriptions = user_data['subscriptions']
+    friend_watched = get_friends_unique_watched(user_data)
+    movie_recommendations = []
+
+    for movie in friend_watched: 
+        if movie['host'] in user_subscriptions: 
+            movie_recommendations.append(movie)
+
+    return(movie_recommendations)
+
+
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    fav_genre = get_most_watched_genre(user_data)
+    friend_watched = get_friends_unique_watched(user_data)
+    movie_recommendations = []
+
+    for movie in friend_watched: 
+        if movie['genre'] == fav_genre: 
+            movie_recommendations.append(movie)
+    return movie_recommendations 
+   
+
+def get_rec_from_favorites(user_data): 
+    user_favorites = user_data['favorites']
+    user_unique = get_unique_watched(user_data)
+    user_recommendations = [] 
+
+    for movie in user_unique:
+        if movie in user_favorites: 

```

Comment:
This might be another place to build an auxiliary set to help with speeding up `in` checks.